src/actions/contractActions.js

```diff
@@ -1,4 +1,16 @@
-export const markdownChanged = markdown => ({
-  type: 'MARKDOWN_CHANGED',
+export const documentEdited = (value, markdown) => ({
+  type: 'DOCUMENT_EDITED',
+  slateValue: value,
   markdown,
 });
+
+export const documentEditedSuccess = (value, markdown) => ({
+  type: 'DOCUMENT_EDITED_SUCCESS',
+  slateValue: value,
+  markdown,
+});
+
+export const documentEditedError = error => ({
```

Comment:
Should we handle this in the reducer? That would already be set up for the `Error` component to handle it in the future.

Reply:
Yes, meant to add that. We should decide how we want to handle these types of errors that are not cicero/ergo errors in the UI.

Reply:
Right. I think this shouldn't hold this PR, and can be a separate issue/feature in the future.